ObjectiveGit/GTIndex.h

```diff
@@ -202,7 +207,53 @@ NS_ASSUME_NONNULL_BEGIN
 ///
 /// Returns `YES` in the event of successful enumeration or no conflicts in the
 /// index, `NO` in case of error.
-- (BOOL)enumerateConflictedFilesWithError:(NSError **)error usingBlock:(void (^)(GTIndexEntry *ancestor, GTIndexEntry *ours, GTIndexEntry *theirs, BOOL *stop))block;
+- (BOOL)enumerateConflictedFilesWithError:(NSError **)error usingBlock:(void (^)(GTIndexEntry * _Nullable ancestor, GTIndexEntry * _Nullable ours, GTIndexEntry * _Nullable theirs, BOOL *stop))block;
+
+/// An enum for use with addPathspecs:flags:error:passingTest: below
+/// See index.h for documentation of each individual git_index_add_option_t flag.
+typedef NS_OPTIONS(NSInteger, GTIndexAddOptionFlags) {
+	GTIndexAddOptionDefault = GIT_INDEX_ADD_DEFAULT,
+	GTIndexAddOptionForce = GIT_INDEX_ADD_FORCE,
+	GTIndexAddOptionDisablePathspecMatch = GIT_INDEX_ADD_DISABLE_PATHSPEC_MATCH,
+	GTIndexAddOptionCheckPathspec = GIT_INDEX_ADD_CHECK_PATHSPEC
+};
+
+/// Add or update index entries matching files in the working directory.
+/// This method will immediately fail if the index's repo is bare.
+///
+/// pathspecs - An `NSString` array of path patterns. (E.g: *.c)
+///             If nil is passed in, all index entries will be updated.
+/// flags     - A combination of GTIndexAddOptionFlags flags
+/// block     - A block run each time a pathspec is matched; before the index is
```

Comment:
"A block that will be called on each match, before the index is changed."
"what the pathspec" => "which pathspec".
"If you pass in ~NULL~ nil"